### PR TITLE
@tboron/preprocessing

### PR DESCRIPTION
**Pull request by t0maboro**
Created layout and some basic types of plots. Choosing columns to reject in progress. If something is wrong with the layout or we should add some other types of plots let me know.

Edit: Layout and plots can be checked if it's ok. Next things I'll realize in order:
- column data integration (date/time format etc.)
- passing a frame of picked columns to the next engine? - or somewhere where it will be reachable for its
- filling missing data with null (next maybe estimate)
- (for now option) automatic dimensions reduction

If I omit something important, let me know.

### CODE REVIEW

**Review by mhawryluk — APPROVED** [review]
Looks great so far!

File: src/widgets/preprocessing_widget.py [hunk 1]
```diff
@@ -1,13 +1,100 @@
...
+    def set_columns_grid(self):
+        columns = self.engine.get_columns()
+        col = max(len(columns) // 11 + 1, 2)
+        rows = (len(columns) - 1) // col + 1
+        self.columns_group.setFixedHeight(min(rows * 60, 350))
+        positions = [(i, j) for i in range(rows) for j in range(col)]
+        for name, position in zip(columns, positions):
+            checkbox = QCheckBox(name)
+            checkbox.setChecked(True)
+            self.columns_grid.addWidget(checkbox, *position)
```
Comment (gregori0o) [inline src/widgets/preprocessing_widget.py:100]: Has the choice of columns some impact to the next step in app? Do you save this choice?

**Review by gregori0o — COMMENTED** [review]
This looks great.

I have a question. How I can get data after preprocessing to use them in algorithm? Is it still "state.imported_data" variables or you create new variables or something else?

**Comment by t0maboro** [comment]
> This looks great.
> 
> I have a question. How I can get data after preprocessing to use them in algorithm? Is it still "state.imported_data" variables or you create new variables or something else?

For now, the plan is to store it in state.imported_data, but this is complex thing to pass it properly. Probably there might be a problem with restoring imported data.

**Comment by t0maboro** [comment]
Right now I used a solution with updating self.state.imported_data. After clicking Select, the program rejects non-checked columns and cuts the dataframe.
Advantages: Dataframe is not duplicated.
Disadvantages: To restore unchecked columns now we probably need to import data once again.

**Pull request merged**